### ISSUE TITLE
kfp.Client does not error out even if host is inaccessible

### ISSUE THREAD

**Issue by ucdmkt** [kubeflow/pipelines#2516]
**What happened:**

kfp.Client did not error out even though gibberish `host`. Instead, `.list_pipeline()` returned empty response.
 
```
$ python
Python 3.5.6 |Anaconda, Inc.| (default, Aug 26 2018, 21:41:56) 
[GCC 7.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> 
>>> 
>>> import kfp
>>> client = kfp.Client(host='foo')
>>> client.list_pipelines()
{'next_page_token': None, 'pipelines': None, 'total_size': None}
>>> print(kfp.__version__)
0.1.32.2
```

**What did you expect to happen:**

either `client = kfp.Client(host='foo')` or `client.list_pipelines()` should have raised runtime error.

**What steps did you take:**
[A clear and concise description of what the bug is.]

```
pip install kfp==0.1.32.2
```

See above for exact code snippet.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Comment by ucdmkt** [comment]
Update: This issue may only be hit iff the environment previously had valid  or invalid? token, possibly due to [this codeline](https://github.com/kubeflow/pipelines/blob/47bb1036f9c558638746bbdccc65034d191d2e8f/sdk/python/kfp/_auth.py#L34)


```
# My environment has gcloud configured, but my current project does not have access to this endpoint.
>>> client = kfp.Client(host='https://xxxxx.notebooks.googleusercontent.com')

# It fails, as expected.
>>> client.list_pipelines()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/google/home/muchida/miniconda3/envs/tfx-kfp-3/lib/python3.5/site-packages/kfp/_client.py", line 286, in list_pipelines
    return self._pipelines_api.list_pipelines(page_token=page_token, page_size=page_size, sort_by=sort_by)
  File "/usr/local/google/home/muchida/miniconda3/envs/tfx-kfp-3/lib/python3.5/site-packages/kfp_server_api/api/pipeline_service_api.py", line 445, in list_pipelines
    (data) = self.list_pipelines_with_http_info(**kwargs)  # noqa: E501
  File "/usr/local/google/home/muchida/miniconda3/envs/tfx-kfp-3/lib/python3.5/site-packages/kfp_server_api/api/pipeline_service_api.py", line 527, in list_pipelines_with_http_info
    collection_formats=collection_formats)
  File "/usr/local/google/home/muchida/miniconda3/envs/tfx-kfp-3/lib/python3.5/site-packages/kfp_server_api/api_client.py", line 330, in call_api
    _preload_content, _request_timeout)
  File "/usr/local/google/home/muchida/miniconda3/envs/tfx-kfp-3/lib/python3.5/site-packages/kfp_server_api/api_client.py", line 161, in __call_api
    _request_timeout=_request_timeout)
  File "/usr/local/google/home/muchida/miniconda3/envs/tfx-kfp-3/lib/python3.5/site-packages/kfp_server_api/api_client.py", line 351, in request
    headers=headers)
  File "/usr/local/google/home/muchida/miniconda3/envs/tfx-kfp-3/lib/python3.5/site-packages/kfp_server_api/rest.py", line 238, in GET
    query_params=query_params)
  File "/usr/local/google/home/muchida/miniconda3/envs/tfx-kfp-3/lib/python3.5/site-packages/kfp_server_api/rest.py", line 228, in request
    raise ApiException(http_resp=r)
kfp_server_api.rest.ApiException: (401)
Reason: Unauthorized



>>> client = kfp.Client(host='foo')
>>> client.list_pipelines()
{'next_page_token': None, 'pipelines': None, 'total_size': None}

**Comment by hongye-sun** [comment]
@IronPan, this looks similar with the issue that we found in last release. I remembered that you fixed in 0.1.32.1. Is it s regression in 0.1.32.2?

**Comment by numerology** [comment]
The problem is as follows.

First kfp.client attempts to determine whether the provided host name is a inverse proxy host or a iap host. If neither is the case, client will believe it's probably a port forward address and set config.host and return. Let me think about what'll be a better way to do this.
